### PR TITLE
Use reflection method to generate encrypt instances

### PR DESCRIPTION
**Pull request by zhujt20**
Use reflection method instead of direct ways to generate encrypt instances.

### CODE REVIEW

File: java/tsfile/src/main/java/org/apache/tsfile/encrypt/AES128Decryptor.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    try {
+      return AES.doFinal(data);
+    } catch (IllegalBlockSizeException | BadPaddingException e) {
+      throw new EncryptException("AES128Decryptor decrypt failed ", e);
+    }
+  }
+
+  @Override
+  public byte[] decrypt(byte[] data, int offset, int size) {
+    //    return decrypt(Arrays.copyOfRange(data, offset, offset + size));
```
Comment (jt2594838) [inline java/tsfile/src/main/java/org/apache/tsfile/encrypt/AES128Decryptor.java]: Remove

File: java/tsfile/src/main/java/org/apache/tsfile/encrypt/AES128Encryptor.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        | InvalidKeyException e) {
+      throw new EncryptException("AES128Encryptor init failed ", e);
+    }
+  }
+
+  @Override
+  public byte[] encrypt(byte[] data) {
+    try {
+      return AES.doFinal(data);
+      //      return AES.doFinal(data);
```
Comment (jt2594838) [inline java/tsfile/src/main/java/org/apache/tsfile/encrypt/AES128Encryptor.java]: Remove, check other places.

File: java/tsfile/src/main/java/org/apache/tsfile/encrypt/EncryptUtils.java [hunk 3]
```diff
@@ -109,37 +158,61 @@ public static IEncryptor getDefaultEncryptor() {
...
-      encryptType = EncryptionType.UNENCRYPTED;
+      encryptType = "org.apache.tsfile.encrypt.UNENCRYPTED";
       dataEncryptKey = null;
     }
-    return IEncryptor.getEncryptor(encryptType, dataEncryptKey);
+    try {
+      Class<?> encryptTypeClass = Class.forName(encryptType);
+      java.lang.reflect.Constructor<?> constructor =
+          encryptTypeClass.getDeclaredConstructor(byte[].class);
+      return ((IEncrypt) constructor.newInstance(dataEncryptKey));
```
Comment (jt2594838) [inline java/tsfile/src/main/java/org/apache/tsfile/encrypt/EncryptUtils.java:173]: The constructor may be stored as a field to reduce the number of reflection calls.

File: java/tsfile/src/main/java/org/apache/tsfile/encrypt/IDecryptor.java [hunk 4]
```diff
@@ -20,47 +20,30 @@
...
-        return new AES128Decryptor(key);
-      default:
-        logger.warn("Unknown encryption type: {}", name);
-        return new NoDecryptor();
+  static IDecryptor getDecryptor(String type, byte[] key) {
+    try {
+      Class<?> encryptClass = Class.forName(type);
+      java.lang.reflect.Constructor<?> constructor =
+          encryptClass.getDeclaredConstructor(byte[].class);
+      return ((IEncrypt) constructor.newInstance(key)).getDecryptor();
```
Comment (jt2594838) [inline java/tsfile/src/main/java/org/apache/tsfile/encrypt/IDecryptor.java]: May try using a concurrent map to store the constructor.

File: java/tsfile/src/main/java/org/apache/tsfile/read/TsFileSequenceReader.java [hunk 5]
```diff
@@ -1721,8 +1721,7 @@ private static ByteBuffer decrypt(IDecryptor decryptor, ByteBuffer buffer) {
       return buffer;
     }
     return ByteBuffer.wrap(
-        decryptor.decrypt(
-            buffer.array(), buffer.arrayOffset() + buffer.position(), buffer.remaining()));
+        decryptor.decrypt(buffer.array(), buffer.position(), buffer.remaining()));
```
Comment (jt2594838) [inline java/tsfile/src/main/java/org/apache/tsfile/read/TsFileSequenceReader.java]: Why is buffer.arrayOffset() removed?

**Review by jt2594838 — APPROVED** [review]

File: java/tsfile/src/main/java/org/apache/tsfile/encrypt/EncryptUtils.java (hunk 3 above)
Comment (zhujt20) [reply java/tsfile/src/main/java/org/apache/tsfile/encrypt/EncryptUtils.java:173]: Actually the constructor here will only be used once.

File: java/tsfile/src/main/java/org/apache/tsfile/encrypt/IDecryptor.java (hunk 4 above)
Comment (zhujt20) [reply java/tsfile/src/main/java/org/apache/tsfile/encrypt/IDecryptor.java]: For decrypt, there may be different kinds of constructors to be used in the future. So maybe storing them is a good idea, but where should I implement this?

Comment (jt2594838) [reply java/tsfile/src/main/java/org/apache/tsfile/encrypt/IDecryptor.java]: Just like the logger above, add a static member.

**Review by jt2594838 — APPROVED** [review]

**Pull request merged**